### ISSUE TITLE
Thrift library headers are under duckdb_apache::thrift but include guards were not updated

### ISSUE THREAD

**Issue by yingsu00** [duckdb/duckdb#3768]
#### What happens?
When adding thrift library to a third party project which uses duckdb Parquet reader, the classes in the standalone Thrift library would be shadowed. This is because the amalgamated file parquet-amalgamation.hpp/cpp copies all Thrift header files, but they are under a non-standard namespace duckdb_apache::thrift. For example, it contains TBase.h like the following:
```
#ifndef _THRIFT_TBASE_H_
#define _THRIFT_TBASE_H_ 1
...
namespace duckdb_apache {
namespace thrift {

class TBase {
public:
  virtual ~TBase() = default;
  virtual uint32_t read(protocol::TProtocol* iprot) = 0;
  virtual uint32_t write(protocol::TProtocol* oprot) const = 0;
};
...
#endif
```
Note that the include guards remain the default names, e.g. `_THRIFT_TBASE_H_` . When we try to add a standalone Thrift library in Velox, these headers shadow the ones in /usr/local/include/thrift because they're all under the same _THRIFT_TBASE_H_ guard. 

Moreover, bitwise_cast() in TProtocol.h is not under the global namespace and would cause double definition problem with the standalone one.

It would be nice to rename the include guards also have "DuckDB" in the string, e.g. `_THRIFT_TBASE_H_` ==> `_DUCKDB_THRIFT_TBASE_H_`. Currently we have workaround PRs to solve this internally, also we temporarily put bitwise_cast() into duckdb_apache::thrift namespace.  But it'll be more clean if DuckDB can do this in the amalgamation script.

**Comment by hannes** [comment]
I think that's a good idea!

**Comment by yingsu00** [comment]
https://github.com/duckdb/duckdb/pull/3999 was submitted to DuckDB.

**Comment by omo** [comment]
(Unsolicited triage): This should be closed (by #3999) @hannes
